homeassistant/components/switch/mochad.py

```diff
@@ -59,17 +65,35 @@ def name(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        self._state = True
         with mochad.REQ_LOCK:
-            self.device.send_cmd('on')
-            self._controller.read_data()
+            try:
+                # Recycle socket on new command to recover mochad connection
+                _LOGGER.debug("Reconnect %s:%s", self._controller.server,
```

Comment:
This should still be move up outside `try... except`. Put as little code as possible within `try... except`, ie only the code that's relevant to catch.

Reply:
Moved it as requested. See commit b1f53449